src/components/CardSelection.tsx

```diff
@@ -0,0 +1,48 @@
+import { memo } from 'react';
+import { CardCompanyType } from '../domain/types';
+
+export interface Props {
+  onChange: (data: object) => void;
+}
+
+const CARD_DATA1 = [
+  { cardCompany: '클린카드1', color: '#ef5350' },
+  { cardCompany: '클린카드2', color: '#ab47bc' },
+  { cardCompany: '클린카드3', color: '#5c6bc0' },
+  { cardCompany: '클린카드4', color: '#29b6f6' },
+];
+const CARD_DATA2 = [
+  { cardCompany: '클린카드5', color: '#26a69a' },
+  { cardCompany: '클린카드6', color: '#9ccc65' },
+  { cardCompany: '클린카드7', color: '#ffee58' },
+  { cardCompany: '클린카드8', color: '#ffa726' },
+];
+
+function CardSelection({ onChange }: Props) {
+  const handleClick = (data: CardCompanyType) => {
+    onChange(data);
+  };
+
+  return (
+    <>
+      <div className="flex-center">
+        {CARD_DATA1.map(({ cardCompany, color }) => (
+          <div className="modal-item-container" key={color} onClick={() => handleClick({ cardCompany, color })}>
+            <div className="modal-item-dot" style={{ backgroundColor: color }}></div>
+            <span className="modal-item-name">{cardCompany}</span>
+          </div>
+        ))}
```

Comment:
useCallback 등으로 메모이제이션이 된 것도 아니기 때문에, `handleClick`이 꼭 필요해보이는 함수로 보이지 않네요! 복잡해지는 포인트는 다음과 같습니다. 

1. 불필요한 CardCompanyType import
2. 구조분해 할당 후 다시 객체 병합

그렇다면 아래는 어떨까요?

```suggestion
        {CARD_DATA1.map((cardInfo) => (
          <div className="modal-item-container" key={cardInfo.color} onClick={() => onChange(cardInfo)}>
            <div className="modal-item-dot" style={{ backgroundColor: cardInfo.color }}></div>
            <span className="modal-item-name">{cardInfo.cardCompany}</span>
          </div>
        ))}
```

혹은 handleClick을 바깥에 분리함으로써 리액트가 props를 비교하는 비용을 조금이나마 줄이고 싶다면, useCallback 등을 사용하는 것이 좋아보여요😊

추가적으로, `object` 타입을 상단에 정의해주셨는데 이런 상위 타입들은 타입을 아주 세밀히 제한하지 못하기 때문에 지양해주시면 좋겠습니다!